### ISSUE TITLE
This may be a bug in R not in Rstudio 

### ISSUE THREAD

**Issue by kemin711** [rstudio/rstudio#8759]
### System details

    RStudio Edition : Server
    RStudio Version : 3.4.3
    OS Version      :  Ubuntu 20
    R Version       : 4.0.3

### Steps to reproduce the problem
Default build of R from source
make install into /usr/local

Then make Rstudio
(fixed a few cmake bugs; otherwise will mot build on my system)

### Describe the problem in detail

Start rstudio-server.service under systemctl
Status fine
Only after tried to login, got linking error

sudo systemctl status rstudio-server | more
● rstudio-server.service - RStudio Server
     Loaded: loaded (/etc/systemd/system/rstudio-server.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2021-01-18 21:31:29 PST; 14min ago
    Process: 3683597 ExecStart=/usr/local/lib/rstudio-server/bin/rserver (code=exited, status=0/SUCCESS)
   Main PID: 3683598 (rserver)
      Tasks: 3 (limit: 309350)
     Memory: 32.2M
     CGroup: /system.slice/rstudio-server.service
             └─3683598 /usr/local/lib/rstudio-server/bin/rserver

Jan 18 21:31:29 myserver_name systemd[1]: Starting RStudio Server...
Jan 18 21:31:29 myserver_name systemd[1]: Started RStudio Server.
Jan 18 21:31:43 myserver_name rsession-kzhou[3683641]: ERROR r error 4 (R code execution error) [errormsg: Error in dyn.load(file, DLLpath = DLLpath, ...) : |||  unable to load shared
 object '/usr/local/lib/R/library/grDevices/libs/grDevices.so':|||  /usr/local/lib/R/library/grDevices/libs/grDevices.so: undefined symbol: Rf_translateCharFP|||]; OCCURRED A
T rstudio::core::Error rstudio::r::exec::{anonymous}::evaluateExpressionsUnsafe(SEXP, SEXP, SEXPREC**, rstudio::r::sexp::Protect*, rstudio::r::exec::{anonymous}::EvalType) sr
c/cpp/r/RExec.cpp:186; LOGGED FROM: int rstudio::r::session::RReadConsole(const char*, CONSOLE_BUFFER_CHAR*, int, int) src/cpp/r/session/RStdCallbacks.cpp:297

More error messages caused by the same root problem

===== checking R information ===
In R 4.0.3 Rf_translateCharFP is a type def

find . -name "*.h" | xargs grep Rf_translate
./include/Rinternals.h:const char * Rf_translateChar(SEXP);
./include/Rinternals.h:const char * Rf_translateChar0(SEXP);
./include/Rinternals.h:const char * Rf_translateCharUTF8(SEXP);
./include/Rinternals.h:#define translateChar		Rf_translateChar
./include/Rinternals.h:#define translateChar0		Rf_translateChar0
./include/Rinternals.h:#define translateCharUTF8      	Rf_translateCharUTF8
./include/Defn.h:const char * Rf_translateCharFP(SEXP);
./include/Defn.h:const char * Rf_translateCharFP2(SEXP);
./include/Defn.h:# define translateCharFP	Rf_translateCharFP
./include/Defn.h:# define translateCharFP2	Rf_translateCharFP2

find . -name "*.c" | xargs grep Rf_translate
find nothing in the R source tree

This is confirmed with the final R library

/usr/local/lib/R/lib$ nm -D libR.so | grep translate
0000000000253620 T Rf_translateChar
0000000000254e90 T Rf_translateChar0
0000000000254ee0 T Rf_translateCharUTF8

### Describe the behavior you expected

Not sure what I did wrong.  Should I use a special tag when building the R language?

- [ ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- I Obtain the latest version of Rstudio by git clone.
- [ ] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kemin711** [comment]
R Defn.h  defined the function,  
  84 const char * Rf_translateCharFP(SEXP);
  85 const char * Rf_translateCharFP2(SEXP);
  86 const char * Rf_trCharUTF8(SEXP);
  87 

But not implemented in any c file.  Maybe this is the source of the trouble.

**Comment by kevinushey** [comment]
Given that RStudio is able to successfully build against R (including recent releases of R) I suspect the issue is more likely a local issue.

On macOS, I see:

```
kevinushey@cdrv-2:/Library/Frameworks/R.framework/Resources/lib
$ nm libR.dylib | grep translate
00000000001a6550 T _Rf_translateChar
00000000001a7ed0 T _Rf_translateChar0
00000000001a57f0 T _Rf_translateCharFP
00000000001a7c40 T _Rf_translateCharFP2
00000000001a7f20 T _Rf_translateCharUTF8
000000000022e970 t __rl_untranslate_macro_value
000000000022b7a0 T _rl_translate_keyseq
000000000022be10 T _rl_untranslate_keyseq
0000000000441510 b _rl_untranslate_keyseq.kseq
00000000001a7830 t _translateToNative
00000000002cfce0 s _yytranslate
00000000002df9c0 s _yytranslate
```

**Comment by kemin711** [comment]
Thanks.  R 4.0.3 defined this function in a header file:void Rf_translateCharFP(); without extern declaration, This function is also not implemented in the full ofR 4.0.3 source.  So could be a library missing at runtime but present at compile time.Do you know which library could contain the function: Rf_translateCharFP()?
Thanks

Kemin Zhou 858 771-3269 

    On Tuesday, January 19, 2021, 10:38:38 AM PST, Kevin Ushey <notifications@github.com> wrote:  
 
 


Given that RStudio is able to successfully build against R (including recent releases of R) I suspect the issue is more likely a local issue.

On macOS, I see:
kevinushey@cdrv-2:/Library/Frameworks/R.framework/Resources/lib
$ nm libR.dylib | grep translate
00000000001a6550 T _Rf_translateChar
00000000001a7ed0 T _Rf_translateChar0
00000000001a57f0 T _Rf_translateCharFP
00000000001a7c40 T _Rf_translateCharFP2
00000000001a7f20 T _Rf_translateCharUTF8
000000000022e970 t __rl_untranslate_macro_value
000000000022b7a0 T _rl_translate_keyseq
000000000022be10 T _rl_untranslate_keyseq
0000000000441510 b _rl_untranslate_keyseq.kseq
00000000001a7830 t _translateToNative
00000000002cfce0 s _yytranslate
00000000002df9c0 s _yytranslate

**Comment by kevinushey** [comment]
It is implemented in R:

https://github.com/wch/r-source/blob/68e5ddfea6e4deda4d8dbc57c570c771883ce62f/src/main/sysutils.c#L1002-L1020
https://github.com/wch/r-source/blob/68e5ddfea6e4deda4d8dbc57c570c771883ce62f/src/include/Defn.h#L1032

R normally remaps non-prefixed function definitions to an `Rf_` prefixed form, hence why you probably didn't see it.
